bugnote_view_inc.php

```diff
@@ -122,6 +122,10 @@
 	$t_can_delete_all_bugnotes = access_has_bug_level( config_get( 'delete_bugnote_threshold' ), $f_bug_id );
 	$t_can_change_view_state_all_bugnotes = $t_can_edit_all_bugnotes && access_has_bug_level( config_get( 'change_view_status_threshold' ), $f_bug_id );
 
+	# Pre-generate form security tokens to avoid performance issues
+	$t_security_token_state = form_security_token( 'bugnote_set_view_state' );
```

Comment:
That is much better than before in terms of perf.  Now we will generate 2 instead of 2n tokens.  You could optimize further by having two functions that get the token.  If the function is called before, it returns the same token.  If the function wasn't called before it creates the token.  Hence, by lazy creation, we could end up generating 0, 1, or maximum 2 tokens.  I'm OK either way, it can be an easy improvement, but it could also be over-engineering :)